state_machine.py

```diff
@@ -410,19 +410,23 @@ def _enter(self, machine):
     def _exit(self, machine):
         pass
 
-    def _update(self, machine, movement):
+    def _update(self, machine):
         # wait 30 seconds for filter to tune in
         now = time.time()
+        if now >= machine.startTime + RESTART_OFFSET_TIME_S:
+            log.info("its time to restart")
+            machine.go_to_state('bootloader')
+            return
+
         if now >= self.enter_timestamp + WAIT_FOR_TUNING_S:
-            if movement:
+            if machine.system.sensor.overshoot: # movement:
```

Comment:
A function in the system class returning the overshoot value would be a cleaner way of doing this; the State Machine shouldn't directly interact with the sensor class.

Reply:
Well, that is kind of tricky, because this is just a flag. But I will have a look